autogluon/task/tabular_prediction/predictor.py

```diff
@@ -688,6 +688,75 @@ def load_data_internal(self, dataset='train', return_X=True, return_y=True):
+        remove_fit_stack : bool, default = True
+            Whether to remove information required to fit new stacking models and continue fitting bagged models with new folds.
+            Only reduces disk usage, it has no impact on memory usage.
+            This includes:
+                out-of-fold (OOF) predictions
+            This is useful for multiclass problems with many classes, as OOF predictions can become very large on disk. (1 GB per model in extreme cases)
+            This disables `predictor.refit_full()` for stacker models.
+        requires_save : bool, default = True
+            Whether to remove information that requires the model to be saved again to disk.
+            Typically this only includes flag variables that don't have significant impact on memory or disk usage, but should technically be updated due to the removal of more important information.
+                An example is the `is_data_saved` boolean variable in `trainer`, which should be updated to `False` if `remove_data=True` was set.
+        reduce_children : bool, default = False
+            Whether to apply the reduction rules to bagged ensemble children models.
+            This should generally be kept as `False` since the most important memory reduction techniques are automatically applied to these models during the original `fit()` call.
+
+        """
+        self._trainer.reduce_memory_size(remove_data=remove_data, remove_fit_stack=remove_fit_stack, remove_fit=True, remove_info=False, requires_save=requires_save, reduce_children=reduce_children)
+
+    def delete_models(self, models_to_keep=None, models_to_delete=None, allow_delete_cascade=False, delete_from_disk=True, dry_run=True):
+        """
+        Deletes models from `predictor`.
+        This can be helpful to minimize memory usage and disk usage, particularly for model deployment.
+        This will remove all references to the models in `predictor`.
+            For example, removed models will not appear in `predictor.leaderboard()`.
+
+        Parameters
+        ----------
+        models_to_keep : str or list, default = None
+            Name of model or models to not delete.
+            All models that are not specified and are also not required as a dependency of any model in `models_to_keep` will be deleted.
+            Specify `models_to_keep='best'` to keep only the best model and its model dependencies.
+            `models_to_delete` must be None if `models_to_keep` is set.
+        models_to_delete : str or list, default = None
+            Name of model or models to delete.
+            All models that are not specified but depend on a model in `models_to_delete` will also be deleted.
+            `models_to_keep` must be None if `models_to_delete` is set.
+        allow_delete_cascade : bool, default = False
+            If `False`, if unspecified dependent models of models in `models_to_delete` exist an exception will be raised instead of deletion occurring.
```

Comment:
provide example of dependent model, e.g. "For example, Model 1 is dependent on Model 2 if Model 1 is a stacker model that takes  predictions from Model 2 as inputs".

Reply:
Added